js/ui/drawer/ui.drawer.js

```diff
@@ -0,0 +1,481 @@
+        this._hideMenuHandler = this.hideMenu.bind(this);
+    },
+
+    _initTemplates() {
+        this.callBase();
+
+        this._defaultTemplates["menu"] = new EmptyTemplate(this);
+        this._defaultTemplates["content"] = new EmptyTemplate(this);
+    },
+
+    _initMarkup() {
+        this.callBase();
+
+        this._togglePositionClass(this.option("menuVisible"));
+        this._renderMarkup();
+
+        this._refreshModeClass();
+        this._refreshShowModeClass();
+
+        const menuTemplate = this._getTemplate(this.option("menuTemplate"));
+
+        menuTemplate && menuTemplate.render({
+            container: this.menuContent()
+        });
+
+        const contentTemplateOption = this.option("contentTemplate"),
+            contentTemplate = this._getTemplate(contentTemplateOption),
+            transclude = this._getAnonymousTemplateName() === contentTemplateOption;
+
+        contentTemplate && contentTemplate.render({
+            container: this.content(),
+            noModel: true,
+            transclude
+        });
+
+        this._initWidth();
+
+        this._renderShader();
+        this._toggleMenuPositionClass();
+    },
+
+    _render() {
+        this.callBase();
+
+        this._dimensionChanged();
+    },
+
+    _renderMarkup() {
+        const $wrapper = $("<div>").addClass(DRAWER_WRAPPER_CLASS);
+        this._$menu = $("<div>").addClass(DRAWER_MENU_CONTENT_CLASS);
+        this._$container = $("<div>").addClass(DRAWER_CONTENT_CLASS);
+
+        $wrapper.append(this._$menu);
+        $wrapper.append(this._$container);
+        this.$element().append($wrapper);
+    },
+
+    _refreshModeClass(prevClass) {
+        prevClass && this.$element()
+            .removeClass(DRAWER_CLASS + "-" + prevClass);
+
+        this.$element().addClass(DRAWER_CLASS + "-" + this.option("mode"));
+    },
+
+    _refreshShowModeClass(prevClass) {
+        prevClass && this.$element()
+            .removeClass(DRAWER_CLASS + "-" + prevClass);
+
+        this.$element().addClass(DRAWER_CLASS + "-" + this.option("showMode"));
+    },
+
+    _renderShader() {
+        this._$shader = this._$shader || $("<div>").addClass(DRAWER_SHADER_CLASS);
+        this._$shader.appendTo(this.content());
+        eventsEngine.off(this._$shader, clickEvent.name);
+        eventsEngine.on(this._$shader, clickEvent.name, this.hideMenu.bind(this));
+        this._toggleShaderVisibility(this.option("menuVisible"));
+    },
+
+    _initWidth() {
+        this._minWidth = this.option("minWidth") || 0;
+        this._maxWidth = this.option("maxWidth") || this.getRealMenuWidth();
+    },
+
+    getMaxWidth() {
+        return this._maxWidth;
+    },
+
+    getMinWidth() {
+        return this._minWidth;
+    },
+
+    getRealMenuWidth() {
+        const $menu = this._$menu;
+        return $menu.get(0).hasChildNodes() ? $menu.get(0).childNodes[0].getBoundingClientRect().width : $menu.get(0).getBoundingClientRect().width;
```

Comment:
$menu.get(0).childNodes[0] - access by index is not reliable and can work incorrectly, see https://trello.com/c/1p1qgIlb